Project.toml

```diff
@@ -33,7 +33,7 @@ NaNMath = "1"
 Preferences = "1"
 SpecialFunctions = "1, 2"
 StaticArrays = "1.5"
-julia = "1.6"
+julia = "1.10"
```

Comment:
Why?

Reply:
No, but if we aren't testing on Julia v1.6 it's best to explicitly drop support. It also allows us to remove weakdeps as deps and use any features from v1.10+.

Reply:
My point is that the package is tested successfully on 1.6, it's that the PR in its current form dropped tests on 1.6 (but adding `min` would fix that). Releasing 1.0 doesn't require making any changes to the package or its dependencies.

Reply:
To be clear, I'm not generally opposed to making such changes but I think they should be done separately. Releasing 1.0 should just involve updating the version number + removing deprecations (if existent) IMO. Otherwise it's also easy to forget to remove VERSION-specific code, old workarounds, etc.

Reply:
I don't really care, push any changes you want to this PR

Reply:
No, I don't want to. I want to keep them in separate PRs. And I don't think any of the changes in this PR apart from the 1.0 bump are necessary for releasing 1.0.

Reply:
Then make your own PR

Reply:
https://github.com/JuliaDiff/ForwardDiff.jl/pull/729

Reply:
https://github.com/JuliaDiff/ForwardDiff.jl/pull/730